### ISSUE TITLE
Intermediate Epoch Validation Metrics

### ISSUE THREAD

**Issue by Gelesh** [keras-team/tf-keras#232]
Especially when each individual data point is large, such as an image data, the batch size need to be small to fit  within the memory. Hence in every epoch there would be too many batch, and each batch would update weights. It may be possible that there is a the validation loss minimum was some where mid way to an end of epoch, and further, the model was mostly overfitting. 
This can be handled by provisioning an intermediate epoch validation matric, like for every batch_ID % n == 0 , where n is configurable, a validation metric is logged.
